src/RepoDriver.sol

```diff
@@ -0,0 +1,530 @@
+    /// `forgeId` is 0 and `nameEncoded` is `name` right-padded with zeros
+    /// When `forge` is GitHub and `name` is longer than 27 bytes,
+    /// `forgeId` is 1 and `nameEncoded` is the lower 27 bytes of the hash of `name`.
+    /// When `forge` is GitLab and `name` is at most 27 bytes long,
+    /// `forgeId` is 2 and `nameEncoded` is `name` right-padded with zeros
+    /// When `forge` is GitLab and `name` is longer than 27 bytes,
+    /// `forgeId` is 3 and `nameEncoded` is the lower 27 bytes of the hash of `name`.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// For GitHub and GitLab it must follow the `user_name/repository_name` structure
+    /// and it must be formatted identically as in the repository's URL,
+    /// including the case of each letter and special characters being removed.
+    /// @return userId The user ID.
+    function calcUserId(Forge forge, bytes memory name) public view returns (uint256 userId) {
+        uint8 forgeId;
+        uint216 nameEncoded;
+        if (forge == Forge.GitHub) {
+            if (name.length <= 27) {
+                forgeId = 0;
+                nameEncoded = uint216(bytes27(name));
+            } else {
+                forgeId = 1;
+                // `nameEncoded` is the lower 27 bytes of the hash
+                nameEncoded = uint216(uint256(keccak256(name)));
+            }
+        } else {
+            if (name.length <= 27) {
+                forgeId = 2;
+                nameEncoded = uint216(bytes27(name));
+            } else {
+                forgeId = 3;
+                // `nameEncoded` is the lower 27 bytes of the hash
+                nameEncoded = uint216(uint256(keccak256(name)));
+            }
+        }
+        // By assignment we get `userId` value:
+        // `zeros (224 bits) | driverId (32 bits)`
+        userId = driverId;
+        // By bit shifting we get `userId` value:
+        // `zeros (216 bits) | driverId (32 bits) | zeros (8 bits)`
+        // By bit masking we get `userId` value:
+        // `zeros (216 bits) | driverId (32 bits) | forgeId (8 bits)`
+        userId = (userId << 8) | forgeId;
+        // By bit shifting we get `userId` value:
+        // `driverId (32 bits) | forgeId (8 bits) | zeros (216 bits)`
+        // By bit masking we get `userId` value:
+        // `driverId (32 bits) | forgeId (8 bits) | nameEncoded (216 bits)`
+        userId = (userId << 216) | nameEncoded;
+    }
+
+    /// @notice Updates the AnyApi operator configuration. Callable only by the admin.
+    /// @param operator The new address of the AnyApi operator.
+    /// @param jobId The new AnyApi job ID used for requesting user owner updates.
+    /// @param defaultFee The new fee in Link for each user owner
+    /// update request when the driver is covering the cost.
+    function updateAnyApiOperator(OperatorInterface operator, bytes32 jobId, uint96 defaultFee)
```

Comment:
We currently don't check of `pending` requests here. This means if the `updateAnyApiOperator` happens between request and `updateOwnerByAnyApi`. The transaction would revert.
Maybe not a big problem but i want to mention it. 

For me the naming is a bit confusion and could be improved `updateAnyApiOperator` `updateOwnerByAnyApi`. 
Completely, different things but overlapping terms.
Maybe:
```solidity
updateOwnerByAnyApi => fulfillUpdateOwner
```

Reply:
That's a good point, we could relatively easily clear the `requestedUpdates` mapping when updating the operator, e.g.
```diff
- mapping(bytes32 requestId => uint256 userId) requestedUpdates;
+ mapping(OperatorInterface operator => mapping(bytes32 requestId => uint256 userId)) requestedUpdates;
```
OTOH what would it protect from? If the operator is malicious, they will steal whatever identity they want anyway. I'm not even sure if removing the `delete storageRef.requestedUpdates[requestId];` line is worth having.

I'm fine with adding the `fulfill` prefix, it's a little verbose, but it's a clear counterpart to `request`. OTOH I'd keep the `ByAnyApi` part, it's meant to cleanly differentiate the API after the update to Chainlink Functions. But maybe it's fine to have clashing names, what bad could happen, and the updated function will probably have different parameters anyway :thinking:

Reply:
It would just mean that some pending requests will fail. We could check this with a counter and only allow a change if no pending requests.  However, then this transaction could be the frontrunner again to block an update.

Reply:
Oh yes, and if there was a single failed request for the repo in the past, e.g. when there was no `FUNDING.json` file yet, it'd be forever blocked.